src/entities/amm.ts

```diff
@@ -1705,51 +1705,19 @@ class AMM {
 
   // underlying token approval for periphery
 
-  public async isUnderlyingTokenApprovedForPeriphery(): Promise<boolean | void> {
-    if (!this.underlyingToken.id) {
-      throw new Error("No underlying token");
+  // token approval for periphery
+  public async isTokenApprovedForPeriphery(tokenAddress: string | undefined): Promise<boolean | void> {
```

Comment:
why don't we enforce tokenAddress to be passed (i.e. tokenAddres: string)?

Reply:
Was just maintaining something closer to the existing interface, which threw an error if there was no underlying token. Can move this error throwing outside the function if you prefer but we may then end up with some duplicated code. wdty?

Reply:
I'd keep this typed to string and check before passing to the function.

Reply:
I think at this stage I would jkeep it where it is just to be sure we are preserving existing behaviour and to prevent me having to retest (I'm technically out of office now until the new year). It's perhaps not what I'd do if this code were going to live forever, but I guess we'll be adjusting some of these interfaces for SDK V2 soon anyway? Thoughts?